sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -937,6 +932,26 @@ class Column(protected[sql] val expr: Expression) extends Logging {
   }
 
   /**
+   * Gives the column a name (alias).
+   * {{{
+   *   // Renames colA to colB in select output.
+   *   df.select($"colA".name("colB"))
+   * }}}
+   *
+   * If the current column has metadata associated with it, this metadata will be propagated
+   * to the new column.  If this not desired, use `as` with explicitly empty metadata.
```

Comment:
Well, actually there is. You can use `@define <name> <definition>` and then reference it using `$name` later. See [here][1]. Scala standard library uses this trick extensively.

[1]: http://docs.scala-lang.org/overviews/scaladoc/for-library-authors.html